javascript/modules/client/src/client-erc20.ts

```diff
@@ -138,41 +223,74 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
     */
   }
 
-  // @ts-ignore  TODO: Remove this comment when implemented
-  private _voteProposal(proposalId: string, vote: VoteOptions) {
-    // TODO: Unimplemented
-    return Promise.reject(new Error("Unimplemented"));
-  }
-  // @ts-ignore  TODO: Remove this comment when implemented
-  private _executeProposal(proposalId: string) {
-    // TODO: Unimplemented
-    return Promise.reject(new Error("Unimplemented"));
+  private async *_voteProposal(_proposalId: string, _vote: VoteOptions): AsyncGenerator<VoteProposalStepValue> {
+    const signer = this.web3.getConnectedSigner();
+    if (!signer) {
+      throw new Error("A signer is needed");
+    } else if (!signer.provider) {
+      throw new Error("A web3 provider is needed");
+    }
+    yield {
+      key: VoteProposalStep.VOTING,
+      txHash: '0x0123456789012345678901234567890123456789012345678901234567890123'
+    }
+    yield {
+      key: VoteProposalStep.DONE,
+      voteId: '0x0123456789012345678901234567890123456789012345678901234567890123'
+    }
   }
-  // private _setDaoConfig(daoAddress: string, config: DaoConfig) {
-  //   return Promise.reject(new Error("Unimplemented"));
-  // }
-  // private _setVotingConfig(daoAddress: string, config: VotingConfig) {
-  //   return Promise.reject(new Error("Unimplemented"));
-  // }
 
-  //// PRIVATE ACTION BUILDER HANDLERS
+  private async *_executeProposal(_proposalId: string): AsyncGenerator<ExecuteProposalStepValue> {
+    const signer = this.web3.getConnectedSigner();
+    if (!signer) {
+      throw new Error("A signer is needed");
+    } else if (!signer.provider) {
+      throw new Error("A web3 provider is needed");
+    }
+    yield {
+      key: ExecuteProposalStep.EXECUTING,
+      txHash: '0x0123456789012345678901234567890123456789012345678901234567890123'
+    }
+    yield {
+      key: ExecuteProposalStep.DONE
+    }
+  }
 
-  // @ts-ignore  TODO: Remove this comment when implemented
-  private _buildActionInit(params: IErc20FactoryParams): FactoryInitParams {
-    // TODO: Unimplemented
-    throw new Error("Unimplemented");
+  private async *_setVotingConfig(_daoAddressOrEns: string, _config: VotingConfig): AsyncGenerator<SetVotingConfigStepValue> {
+    const signer = this.web3.getConnectedSigner();
+    if (!signer) {
+      throw new Error("A signer is needed");
+    } else if (!signer.provider) {
+      throw new Error("A web3 provider is needed");
+    }
+    yield {
+      key: SetVotingConfigStep.CONFIGURING,
```

Comment:
If this function will be creating a new proposal, then I'd name this step as `CREATING_PROPOSAL` instead.